sys/xtimer/xtimer_core.c

```diff
@@ -480,13 +479,18 @@ static void _timer_callback(void)
 
         /* fire timer */
         _shoot(timer);
+
+        /* time update after executing callback */
+        if (!timer->trivial_callback) {
+          now = _xtimer_lltimer_now();
```

Comment:
for instance, keep track (e.g. `boolean non-trivial = false`) if at least one callback was non-trivial and than simplify below to 

```
if (non-trivial && (reference > _xtimer_lltimer_now()) {
```